Change profile claim to be URL or OID (#93)
ietf-rats-wg/eat: draft-ietf-rats-eat.md

    - fullname: B. de Medeiros
    - fullname: C. Mortimore

>   CBOR-OID:
>     target: https://tools.ietf.org/html/draft-ietf-cbor-tags-oid-05
>     title: Concise Binary Object Representation (CBOR) Tags for Object Identifiers
>     date: 2021
>     author: 
>     - fullname: C. Bormann

Reviewer: ```suggestion
  CBOR-OID: I-D.ietf-cbor-tags-oid
```

Reviewer: so that updates to the ref are tracked automatically

## The Profile Claim (profile) {#profile-claim}

A profile is identified by either a URL or an OID.
Typically, the URL will reference a document describing the profile.
An OID is just a unique identifier for the profile.

> It may exist in anywhere in the OID tree.

Reviewer: ```suggestion
It may exist anywhere in the OID tree.
```

> There is no requirement that the named document be publicly accessible.

Reviewer: I think this should be moved up to L959

## The Profile Claim (profile) {#profile-claim}

See {{profiles}} for the detailed description of a profile.

> A profile is identified by either a URL or an OID.

Reviewer: ```suggestion
A profile is identified by either an absolute URI {{!RFC3896}} or an absolute OID (ASN.1 Type "OBJECT IDENTIFIER").
```

> Typically, the URL will reference a document describing the profile.

Reviewer: ```suggestion
Typically, the URI will reference a document describing the profile.
```

> An OID is just a unique identifier for the profile.
> It may exist anywhere in the OID tree.

Reviewer: ```suggestion
```
these two lines are duplicates of L973-974

There is no requirement that the named document be publicly accessible.
An OID is just a unique identifier for the profile.

> It may exist anywhere in the OID tree.

Reviewer: I don't understand why is it important to state this?

The primary purpose of the profile claim is to uniquely identify the profile even if it is a private profile.

The OID is encoded in CBOR according to {{CBOR-OID}} and the URI according to {{RFC8949}}.

> Both are unwrapped and thus not tags.

Reviewer: ```suggestion
Both are unwrapped and thus the outer tag is removed.
```

Author: I'm leaving this as is because, there is no possibility of any tag here (no inner tag). Also, contrary to normal English use of the word tag, in CBOR the tag is the whole thing with the value, not just the tag number.

* bstr -- must be base64url encoded
* time -- must be encoded as NumericDate as described section 2 of {{RFC7519}}.
* string-or-uri -- must be encoded as StringOrURI as described section 2 of {{RFC7519}}.
* uri -- must be a URI {{RFC3986}}.

> * oid -- is encoded as a string represetation of the OID digits.

Reviewer: ```suggestion
* oid -- is encoded using dotted decimal notation as per Section 5 of {{CBOR-OID}} 
```

Author: This is JSON representation of OIDs so a reference to CBOR representation of OID seems wrong.
I haven't found any standard to reference for text representation of OIDs in JSON so far.

Reviewer: Yeah that doesn't sound entirely right, but it's the only reference I had found.

I've given my google-foo another spin and found that Section 3.2 of RFC5675 has an ABNF for the dotted-decimal notation.  Or otherwise let's just say "using the well established dotted-decimal notation" or similar

Author: OK. Pushed a change for this.
